src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -2134,6 +2134,12 @@ def _find_managed_instance_sku_from_capabilities(cli_ctx, location, sku):
     # Find family level capability, based on requested sku properties
     family_capability = _find_family_capability(sku, edition_capability.supported_families)
 
+    # Check whether hardware family is being changed to a deprecated hardware family
+    if not is_instance_create:
+        if current_sku_family == 'Gen5' and 'Gen4' in family_capability.sku:
```

Comment:
Why is this being done client-side instead of server-side (MI provision operation authorizer)? These sorts of policy decisions should not be coded into CLI

Reply:
The idea was to fast fail if a customer tries to change hardware family to a deprecated hardware family. Synced with PM, and we agreed to remove this check as you suggested. Thanks